zerver/webhooks/github/view.py

```diff
@@ -103,9 +105,12 @@ def get_assigned_or_unassigned_pull_request_body(helper: Helper) -> str:
         number=pull_request["number"].tame(check_int),
         title=pull_request["title"].tame(check_string) if include_title else None,
     )
-    if assignee:
-        return base_message.replace("assigned", f"assigned {stringified_assignee} to", 1)
-    return base_message
+
+    action = payload["action"].tame(check_string)
+    assignee = payload["assignee"]["login"].tame(check_string)
+    replacement_template = REPLACEMENTS_IN_ASSIGNMENT_TEMPLATE[action].format(assignee=assignee)
+
+    return base_message.replace(action, replacement_template, 1)
```

Comment:
Reading this, why do we need to do this `.replace` hackery? It seems fragile (what if the word "assignee" is in the PR title?) and I'd have thought we could just pass the right parameters into `get_pull_request_event_message` to do assignees?

Reply:
True, nice catch, thank you! Replaced that 'replace'.

I was actually in the middle of typing out just that - regarding the `assignee` values being different, and that we could refactor using  `get_pull_request_event_message` since all the other message templates are there, and why I wasn't sure and didn't yet, as self-review comments.

Among the integrations that use that function (Bitbucket, Gitlab, Gitea, Gogs, etc), I noticed that only GitHub supports the unassignment events. So, I implemented it locally in the GitHub webhook's view. Also, it looked like a much heavier refactor to me, than it turned out.

I've now updated the library function `get_pull_request_event_message` and all the webhook handlers that support assignment/unassignment events.